File: cs101/pa2/Lex.c
//Kevin Lee
//1480757
//pa2
//Cmps101
//Lex.java

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "List.h"
#define MAX_LEN 160

int main (int argc, const char * argv[]) {
    FILE *in, *out, *in2;
    char line[MAX_LEN];

        int lineNumber = 0;
        int i = 0;

        if( argc != 3 ){
        printf("Usage: %s infile outfile\n Bad Error", argv[0]);
        exit(1);
    }
       
        in = fopen(argv[1], "r");
        out = fopen(argv[2], "w");
        in2 = fopen(argv[1], "r");

        while( fgets(line, MAX_LEN, in) != NULL){
        lineNumber++;
        }
        fclose(in);
        
        
        char** list = malloc(sizeof(char**) * lineNumber);
        while( fgets(line,MAX_LEN, in2) != NULL){
            char *D = malloc(strlen(line) + 1);
            strcpy(D,line);
            list[i]= D;
            i++;
        }
        fclose(in2);

        List L = newList();
        append(L,0);
	//printf("%i", lineNumber);
        for(i=1; i<lineNumber; i++){
            char *temp = list[i];
            moveFront(L);
		if (strcmp(temp,list[back(L)])>0){
                        append(L,i);
                }
                else if (strcmp(temp,list[front(L)])<0){
                        prepend(L,i);
                }
                else if(index(L) != -1){
                       while(strcmp(temp,list[get(L)])>0){
			                   //printf("loop");
                        moveNext(L);
                }
        insertBefore(L,i);

        }
     }

        for(moveFront(L); index(L)>=0; moveNext(L)){
            fprintf(out, "%s", list[get(L)]);
        }
        fclose(out);

    for(int i = 0; i < lineNumber; i++){
    free(list[i]);
  }
  free(list);
  freeList(&L);
}

File: cs101/pa1/Lex.java
//Kevin Lee
//1480757
//pa1
//Cmps101
//Lex.java

import java.io.*;
import java.util.Scanner;

class Lex{
    public static void main(String[] args) throws IOException{
        Scanner in = null;
        Scanner in2 = null;
        PrintWriter out = null;
        String line = null;
        String[] token = null;
        int n, lineNumber = 0;
        String[] ArrayOne;
        List list = new List();
        int i = 0;

        if (args.length != 2){
            System.err.println ("Lex infile outfile");
            System.exit(1);
        }

        in = new Scanner(new File(args[0]));
        in2 = new Scanner(new File(args[0]));
        out = new PrintWriter(new FileWriter(args[1]));

        while( in.hasNextLine() ){
            lineNumber++;
            in.nextLine();
        }
        in.close();
        ArrayOne = new String[lineNumber];
        while(in2.hasNextLine()){      
            ArrayOne[i]=  in2.nextLine();
            i++;
        }
	in2.close();

	list.prepend(0);
	//System.out.println(lineNumber);
        for(i=1; i<lineNumber; i++){ 
            String temp = ArrayOne[i];
	    list.moveFront();
		if (temp.compareTo(ArrayOne[list.back()])>0){
                        list.append(i);
                }
                else if (temp.compareTo(ArrayOne[list.front()])<0){
                        list.prepend(i);
                }
                else if(list.index() != -1){
                        while(temp.compareTo(ArrayOne[list.get()])>0){
                        //System.out.println("loop");
                        list.moveNext();
                }
        list.insertBefore(i);

	}
     }
       
        for(list.moveFront(); list.index()>=0; list.moveNext()){
            out.println(ArrayOne[list.get()]+" ");
        }
        out.close();
    }


}
